flake.nix

```diff
@@ -289,7 +289,7 @@
         workspaceDeps = craneLib.buildDepsOnly (commonArgs // {
           src = filterWorkspaceDepsBuildFiles ./.;
           pname = "workspace-deps";
-          buildPhaseCargoCommand = "cargo doc && cargo check --profile release --all-targets && cargo build --profile release --all-targets";
+          buildPhaseCargoCommand = "cargo doc --profile $CARGO_PROFILE && cargo check --profile $CARGO_PROFILE --all-targets && cargo build --profile $CARGO_PROFILE --all-targets";
```

Comment:
BTW. Lack of `--profile $CARGO_PROFILE` for `cargo doc` might have resulted in building docs in debug mode, which might have led to larger output file sizes.